samza-core/src/main/java/org/apache/samza/runtime/ApplicationRunnerMain.java

```diff
@@ -43,9 +49,33 @@ public ApplicationRunnerOperation getOperation(OptionSet options) {
       String rawOp = options.valueOf(operationOpt);
       return ApplicationRunnerOperation.fromString(rawOp);
     }
+
+    @Override
+    public Config loadConfig(OptionSet options) {
```

Comment:
Do you feel that the "base" use case for `CommandLine` is to need the full config? To me, it looks a little awkward to need to override `loadConfig` here with an implementation which is almost the same as the base `loadConfig`.
I suppose one alternative would be to make this impl here the base impl, and then have each use case call `ConfigUtil.loadConfig` if is needs it. What do you see as the tradeoffs?

Reply:
Good question, in all 13 direct usage of CommandLine, and 5 sub classes of CommandLine, they are all expecting full job configs to work properly, e.g. RocksDbReadingTool extends CommandLine to read full job config before being able to read data from Rocks DB, which is being invoked by CLI.

If we change CommandLine to the other way, that means we need to invoke ConfigUtil.loadConfig explicitly in 13 places, and override loadConfigs in 5 sub classes to invoke ConfigUtil.loadConfig too, compared to overriding once in ApplicationRunnerCommandLine only.

I would prefer the current way to keep the base class to serve majority use case.